K2Field.K2NE.ServiceBroker/ServiceObjects/PowerShell/DynamicPowerShellSO.cs

```diff
@@ -0,0 +1,112 @@
+﻿using K2Field.K2NE.ServiceBroker.Helpers.PowerShell;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using SourceCode.SmartObjects.Services.ServiceSDK.Objects;
+using SourceCode.SmartObjects.Services.ServiceSDK.Types;
+using K2Field.K2NE.ServiceBroker.Helpers;
+using System.Data;
+
+namespace K2Field.K2NE.ServiceBroker.ServiceObjects.PowerShell
+{
+    public class DynamicPowerShellSO : ServiceObjectBase
+    {
+        public DynamicPowerShellSO(K2NEServiceBroker api)
+            : base(api)
+        {
+        }
+
+        public override string ServiceFolder
+        {
+            get
+            {
+                return Constants.ServiceFolders.PowerShell;
+            }
+        }
+
+        public string PowerShellSubdirectories
+        {
+            get
+            {
+                if (!ServiceBroker.Service.ServiceConfiguration.Contains(Constants.ConfigurationProperties.PowerShellSubdirectories))
+                {
+                    return String.Empty;
+                }
+                return Convert.ToString(ServiceBroker.Service.ServiceConfiguration[Constants.ConfigurationProperties.PowerShellSubdirectories]);
+            }
+        }
+
+        public override List<SourceCode.SmartObjects.Services.ServiceSDK.Objects.ServiceObject> DescribeServiceObjects()
+        {
+            List<ServiceObject> serviceObjects = new List<ServiceObject>();
+
+            if(!String.IsNullOrEmpty(PowerShellSubdirectories))
```

Comment:
Commented it in previous part. I think that it's a correct behavior. If we have an empty string in this property that mean that we don't use any folders.